### ISSUE TITLE
Compiler crash when breaking from runtime loop at comptime

### ISSUE THREAD

**Issue by mlugg** [ziglang/zig#20588]
### Zig Version

0.14.0-dev.224+95d9292a7

### Steps to Reproduce and Observed Behavior

```zig
export fn f() void {
    while (true) {
        comptime break;
    }
}
```
This specific example doesn't crash a ReleaseFast compiler, but ("correctly") crashes a debug compiler. More complex examples can probably crash the ReleaseFast compiler.

### Expected Behavior

The `comptime break` here should emit an AstGen error. Specifically, when encountering a `break` or `continue`, AstGen should compare `gz.is_comptime` to `is_comptime` for the loop `GenZir`, and emit an error if they don't match.

**Comment by fmaggi** [comment]
Are `comptime break` and `comptime continue` ever valid? In a comptime scope you can just wirte `break` or `continue`, right?

**Comment by mlugg** [comment]
Yes.

**Comment by fmaggi** [comment]
So `comptime break` and `comptime continue` should always be a compile error?

**Comment by mlugg** [comment]
Yes. The error mechanism I described in the original post achieves that.

**Comment by fmaggi** [comment]
Ok, I was making sure I understood correctly
